src/xref2/ref_tools.ml

```diff
@@ -232,44 +246,42 @@ module CL = struct
   let of_element _env (`Class (id, t)) : t = (`Identifier id, t)
 
   let in_env env name =
-    env_lookup_by_name Env.s_class name env >>= fun e -> Some (of_element env e)
+    env_lookup_by_name Env.s_class name env >>= fun e -> Ok (of_element env e)
 
-  let of_component _env c ~parent_ref name : t option =
-    Some (`Class (parent_ref, name), c)
+  let of_component _env c ~parent_ref name = Ok (`Class (parent_ref, name), c)
 end
 
 module CT = struct
-  type t = class_type_lookup_result
+  (* type t = class_type_lookup_result *)
```

Comment:
For these I'm quite tempted to keep the type here as documentation (internal right now, since it's not actually exposed in the interface) - though the alias is not terribly helpful. How about:

```ocaml
type t = Resolved.ClassType.t * Component.ClassType.t
```

It also might be useful to declare the error type too - again, mainly for documentation purposes. What do you think?

Reply:
I'd too prefer to define this type fully and define `class_type_lookup_result` as alias to it but I prefer to not duplicate the definition.
I've added back all the types and added an instance of the result type to the first function of every modules (for documentation).

Reply:
I don't think we need `class_type_lookup_result` - so no duplication needed. I think the only one actually exposed is `module_lookup_result`

Reply:
(beyond the fact that they are used in subsequent result types, which we can duplicate in the modules I think)

Reply:
It's not exposed but it's used to define `type_lookup_result` and that's the duplication I would like to avoid. I think it's also useful to show that `type_lookup_result` is an aggregate of existing return types rather than a new thing.

Reply:
OK, let's keep this as-is for now, we might still be able to tidy up but it's not a big issue :-)